docs/reference/part-environment-variables.rst

```diff
@@ -3,4 +3,139 @@
 Part environment variables
 ==========================
 
-.. inherit from Craft Parts
+When building a part to construct a snap, Snapcraft creates the following sets of
+environment variables that can optionally be used by a part's build mechanism:
+
+
+Directories
+-----------
+
+See :ref:`explanation-parts-lifecycle` for details on which directories are created when
+building a part.
+
+
+core22 | core24
+~~~~~~~~~~~~~~~
+
+Snapcraft exposes the following directory-related environment variables when building with
+core22 and core24:
+
+.. list-table::
+
+    * - ``CRAFT_PART_SRC``
+
+        ``SNAPCRAFT_PART_SRC``
+      - The absolute path to where a part's sources are pulled. It's also the part's
+        working directory for the pull step.
+    * - ``CRAFT_PART_SRC_WORK``
+
+        ``SNAPCRAFT_PART_SRC_WORK``
+      - The absolute path to the part source subdirectory, if any. Defaults to the part
+        source directory
+    * - ``CRAFT_PART_BUILD``
+
+        ``SNAPCRAFT_PART_BUILD``
+      - The absolute path to the sources used for the part's build step. It is also the
+        working directory of the build step.
+    * - ``CRAFT_PART_BUILD_WORK``
+
+        ``SNAPCRAFT_PART_BUILD_WORK``
+      - The absolute path to the part build subdirectory in case of out-of-tree builds.
+        Defaults to the part source directory.
+    * - ``CRAFT_PART_INSTALL``
+
+        ``SNAPCRAFT_PART_INSTALL``
+      - The absolute path to the results of the part's build step. It also contains the
+        staged packages of that part.
+    * - ``CRAFT_PRIME``
+
+        ``SNAPCRAFT_PRIME``
+      - The absolute path to where files are primed.
+    * - ``CRAFT_PROJECT_DIR``
+
+        ``SNAPCRAFT_PROJECT_DIR``
+      - The absolute path to the root of the Snapcraft project.
+    * - ``CRAFT_STAGE``
+
+        ``SNAPCRAFT_STAGE``
+      - The absolute path to where files are staged.
+    * - ``CRAFT_OVERLAY``
+      - The absolute path for the part's layer directory during the overlay step if
+        overlays are enabled.
```

Comment:
Can you drop this? I don't think it's available for snapcraft, but even if it is, it shouldn't be documented since Snapcraft doesn't use overlays.